river/command/keyboard.zig

```diff
@@ -58,15 +72,30 @@ pub fn keyboardLayout(
     ) orelse return error.InvalidValue;
     defer new_keymap.unref();
 
-    server.config.keymap.unref();
-    server.config.keymap = new_keymap.ref();
+    try applyLayout(new_keymap);
+}
 
-    var it = server.input_manager.devices.iterator(.forward);
-    while (it.next()) |device| {
-        if (device.wlr_device.type != .keyboard) continue;
-        const wlr_keyboard = device.wlr_device.toKeyboard();
-        // wlroots will log an error if this fails and there's unfortunately
-        // nothing we can really do in the case of failure.
-        _ = wlr_keyboard.setKeymap(new_keymap);
-    }
+pub fn keyboardLayoutFile(
+    _: *Seat,
+    args: []const [:0]const u8,
+    _: *?[]const u8,
+) Error!void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+
+    var file = std.fs.cwd().openFile(args[1], .{}) catch return error.CannotOpenFile;
+    defer file.close();
+
+    var file_bytes = file.readToEndAlloc(gpa.allocator(), 512 * 4096) catch return error.CannotReadFile;
```

Comment:
This operation has to relevant failures: Either the file isn't readable, or we run out of memory.
```suggestion
    var file_bytes = file.readToEndAlloc(gpa.allocator(), 512 * 4096) catch |err| {
        switch(err) {
             error.OutOfMemory => return err,
            else => return error.CannotReadFile,
        }
    };
```